src/ggml-rpc.cpp

```diff
@@ -0,0 +1,622 @@
+#include "ggml-rpc.h"
+#include "ggml.h"
+#include "ggml-backend-impl.h"
+#include "ggml-rpc.grpc.pb.h"
+
+#ifdef GGML_USE_CUBLAS
+#include "ggml-cuda.h"
+#endif
+
+#include <grpcpp/grpcpp.h>
+#include <memory>
+#include <string>
+
+#define UNUSED GGML_UNUSED
+
+#define GGML_DEBUG 1
+#if (GGML_DEBUG >= 1)
+#define GGML_PRINT_DEBUG(...) printf(__VA_ARGS__)
+#else
+#define GGML_PRINT_DEBUG(...)
+#endif
+
+static ggml_guid_t ggml_backend_rpc_guid() {
+    static ggml_guid guid = { 0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88, 0x99, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff};
+    return &guid;
+}
+
+struct ggml_backend_rpc_context {
+    std::string endpoint;
+    std::string name;
+    std::shared_ptr<ggml::Backend::Stub> stub;
+};
+
+// rpc buffer
+
+struct ggml_backend_rpc_buffer_context {
+    std::shared_ptr<ggml::Backend::Stub> stub;
+    uint64_t remote_ptr;
+    std::string name;
+};
+
+GGML_CALL static const char * ggml_backend_rpc_buffer_get_name(ggml_backend_buffer_t buffer) {
+    GGML_PRINT_DEBUG("buffer get name\n");
+    ggml_backend_rpc_buffer_context * ctx = (ggml_backend_rpc_buffer_context *)buffer->context;
+    return ctx->name.c_str();
+}
+
+GGML_CALL static void ggml_backend_rpc_buffer_free_buffer(ggml_backend_buffer_t buffer) {
+    GGML_PRINT_DEBUG("buffer free\n");
+    ggml_backend_rpc_buffer_context * ctx = (ggml_backend_rpc_buffer_context *)buffer->context;
+    ggml::FreeBufferRequest request;
+    request.set_bufptr(ctx->remote_ptr);
+    ggml::FreeBufferReply reply;
+    grpc::ClientContext context;
+    grpc::Status status = ctx->stub->FreeBuffer(&context, request, &reply);
+    GGML_ASSERT(status.ok());
+}
```

Comment:
Unless it is done somewhere else, this function should also free `buffer->context`.

Reply:
Fixed, thanks!